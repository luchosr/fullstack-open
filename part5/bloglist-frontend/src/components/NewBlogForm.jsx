import React, { useState } from 'react';
import PropTypes from 'prop-types';

const NewBlogForm = ({ handleSubmit }) => {
  const [newBlogTitle, setNewBlogTitle] = useState('');
  const [newBlogAuthor, setNewBlogAuthor] = useState('');
  const [newBlogUrl, setNewBlogUrl] = useState('');

  return (
    <div>
      <h2>Create new Blog</h2>
      <form
        className="blog-form"
        onSubmit={(event) => {
          event.preventDefault();
          handleSubmit(newBlogTitle, newBlogAuthor, newBlogUrl);
          setNewBlogAuthor('');
          setNewBlogTitle('');
          setNewBlogUrl('');
        }}
      >
        <div>
          Title:
          <input
            id="blogTitle"
            type="text"
            value={newBlogTitle}
            placeholder="add a blog title here"
            onChange={({ target }) => setNewBlogTitle(target.value)}
          />
        </div>
        <div>
          Author:
          <input
            id="blogAuthor"
            type="text"
            value={newBlogAuthor}
            placeholder="add a blog author here"
            onChange={({ target }) => setNewBlogAuthor(target.value)}
          />
        </div>
        <div>
          Url:
          <input
            id="blogUrl"
            type="text"
            value={newBlogUrl}
            placeholder="add a blog url here"
            onChange={({ target }) => setNewBlogUrl(target.value)}
          />
        </div>
        <button id="createNewBlogButton" type="submit">
          create
        </button>
      </form>
    </div>
  );
};

NewBlogForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default NewBlogForm;
