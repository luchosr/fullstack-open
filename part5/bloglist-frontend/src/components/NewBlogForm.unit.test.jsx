import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NewBlogForm from './NewBlogForm';
import { input } from '@testing-library/user-event/dist/cjs/event/input.js';

test.only('<NewBlogForm /> updates parent state and calls onSubmit', async () => {
  const submitBlog = vi.fn();
  const user = userEvent.setup();

  render(<NewBlogForm handleSubmit={submitBlog} />);

  const titleInput = screen.getByPlaceholderText('add a blog title here');
  const authorInput = screen.getByPlaceholderText('add a blog author here');
  const urlInput = screen.getByPlaceholderText('add a blog url here');

  const sendButton = screen.getByText('create');

  await user.type(titleInput, 'testing a form... title');
  await user.type(authorInput, 'testing the author');
  await user.type(urlInput, 'testingurl.com');

  await user.click(sendButton);

  expect(submitBlog.mock.calls).toHaveLength(1);
  expect(submitBlog.mock.calls[0][0]).toBe('testing a form... title');
  expect(submitBlog.mock.calls[0][1]).toBe('testing the author');
  expect(submitBlog.mock.calls[0][2]).toBe('testingurl.com');
});
