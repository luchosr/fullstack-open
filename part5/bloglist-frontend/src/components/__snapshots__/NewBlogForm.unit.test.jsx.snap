// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NewBlogForm /> should pass a snapshot testing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Create new Blog
        </h2>
        <form>
          <div>
            Title:
            <input
              placeholder="add a blog title here"
              type="text"
              value=""
            />
          </div>
          <div>
            Author:
            <input
              placeholder="add a blog author here"
              type="text"
              value=""
            />
          </div>
          <div>
            Url:
            <input
              placeholder="add a blog url here"
              type="text"
              value=""
            />
          </div>
          <button
            type="submit"
          >
            create
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Create new Blog
      </h2>
      <form>
        <div>
          Title:
          <input
            placeholder="add a blog title here"
            type="text"
            value=""
          />
        </div>
        <div>
          Author:
          <input
            placeholder="add a blog author here"
            type="text"
            value=""
          />
        </div>
        <div>
          Url:
          <input
            placeholder="add a blog url here"
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          create
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<NewBlogForm /> updates parent state and calls onSubmit 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Create new Blog
        </h2>
        <form>
          <div>
            Title:
            <input
              placeholder="add a blog title here"
              type="text"
              value=""
            />
          </div>
          <div>
            Author:
            <input
              placeholder="add a blog author here"
              type="text"
              value=""
            />
          </div>
          <div>
            Url:
            <input
              placeholder="add a blog url here"
              type="text"
              value=""
            />
          </div>
          <button
            type="submit"
          >
            create
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Create new Blog
      </h2>
      <form>
        <div>
          Title:
          <input
            placeholder="add a blog title here"
            type="text"
            value=""
          />
        </div>
        <div>
          Author:
          <input
            placeholder="add a blog author here"
            type="text"
            value=""
          />
        </div>
        <div>
          Url:
          <input
            placeholder="add a blog url here"
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          create
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`new blog form should pass a snapshot testing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Create new Blog
        </h2>
        <form>
          <div>
            Title:
            <input
              placeholder="add a blog title here"
              type="text"
              value=""
            />
          </div>
          <div>
            Author:
            <input
              placeholder="add a blog author here"
              type="text"
              value=""
            />
          </div>
          <div>
            Url:
            <input
              placeholder="add a blog url here"
              type="text"
              value=""
            />
          </div>
          <button
            type="submit"
          >
            create
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Create new Blog
      </h2>
      <form>
        <div>
          Title:
          <input
            placeholder="add a blog title here"
            type="text"
            value=""
          />
        </div>
        <div>
          Author:
          <input
            placeholder="add a blog author here"
            type="text"
            value=""
          />
        </div>
        <div>
          Url:
          <input
            placeholder="add a blog url here"
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          create
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
